docs/resources/packer_channel.md

```diff
@@ -0,0 +1,111 @@
+subcategory: ""
+description: |-
+  The Packer Channel resource allows you to manage image bucket channels within an active HCP Packer Registry.
+---
+
+# hcp_packer_channel (Resource)
+
+The Packer Channel resource allows you to manage image bucket channels within an active HCP Packer Registry.
+
+## Example Usage
+
+To create a channel with no assigned iteration.
+```terraform
+resource "hcp_packer_channel" "staging" {
+  name        = "staging"
+  bucket_name = "alpine"
+}
+```
+
+To create, or update an existing, channel with an assigned iteration.
+```terraform
+resource "hcp_packer_channel" "staging" {
+  name        = "staging"
+  bucket_name = "alpine"
+  iteration_assignment {
+    id = "iteration-id"
+  }
+}
+
+# Update assigned iteration using an iteration fingerprint
+resource "hcp_packer_channel" "staging" {
+  name        = "staging"
+  bucket_name = "alpine"
+  iteration_assignment {
+    fingerprint = "fingerprint-associated-to-iteration"
+  }
+}
+
+# Update assigned iteration using an iteration incremental version
+resource "hcp_packer_channel" "staging" {
+  name        = "staging"
+  bucket_name = "alpine"
+  iteration_assignment {
+    // incremental_version is the version number assigned to a completed iteration.
+    incremental_version = 1
+  }
+}
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `bucket_name` (String) The slug of the HCP Packer Registry image bucket where the channel should be managed in.
+- `name` (String) The name of the channel being managed.
+
+### Optional
+
+- `iteration_assignment` (Block List, Max: 1) The iteration assignment information that will be used to assign a completed iteration to the channel. (see [below for nested schema](#nestedblock--iteration_assignment))
+- `timeouts` (Block, Optional) (see [below for nested schema](#nestedblock--timeouts))
+
+### Read-Only
+
+- `author_id` (String) The author of the channel.
+- `created_at` (String) Creation time of this build.
+- `id` (String) The ID of this resource.
+- `iteration` (List of Object) The iteration assigned to the channel. (see [below for nested schema](#nestedatt--iteration))
+- `organization_id` (String) The ID of the organization this HCP Packer registry is located in.
+- `project_id` (String) The ID of the project this HCP Packer registry is located in.
+
+<a id="nestedblock--iteration_assignment"></a>
+### Nested Schema for `iteration_assignment`
+
+Optional:
+
+- `fingerprint` (String) The fingerprint of the iteration to assign to the channel.
+- `id` (String) The id of the iteration to assign to the channel.
+- `incremental_version` (Number) The incremental_version of the iteration to assign to the channel.
+
+
+<a id="nestedblock--timeouts"></a>
+### Nested Schema for `timeouts`
+
+Optional:
+
+- `create` (String)
+- `default` (String)
+- `delete` (String)
+- `update` (String)
+
+
+<a id="nestedatt--iteration"></a>
+### Nested Schema for `iteration`
+
+Read-Only:
+
+- `fingerprint` (String)
```

Comment:
I wonder why these are not including the description you gave them 🤔

Reply:
@bcmdarroch Any thoughts on why the docs don't get generated for these nested attributes?

Reply:
Yeah this is a known issue: https://github.com/hashicorp/terraform-plugin-docs/issues/28. Unfortunately can't add further detail. :( The provider SDK doesn't have great support for nested objects in general. 

Not sure if this would resolve this specific doc issue, but the new plugin framework has much better support for nested objects: https://developer.hashicorp.com/terraform/plugin/framework/migrating/benefits